File: server/Models/Recipe.cs
using System.ComponentModel.DataAnnotations;

namespace spice.Models
{
    public class Recipe
    {
        public int Id { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        [MaxLength(500)] public string Title { get; set; }
        [MaxLength(500)] public string Instructions { get; set; }
        [MaxLength(1000)] public string img { get; set; }
        [MaxLength(500)] public string Category { get; set; }
        public string CreatorId { get; set; }
        public Profile Creator { get; set; }
    }
}

File: client/src/models/Recipe.js
export class Recipe {
    constructor(data) {
        this.id = data.id
        this.createdAt = data.createdAt ? new Date(data.createdAt) : null
        this.updatedAt = data.updatedAt ? new Date(data.updatedAt) : null
        this.title = data.title
        this.instructions = data.instructions
        this.img = data.img
        this.category = data.category
        this.creatorId = data.creatorId
        this.creator = data.creator
    }
}